builds/misc/mqtt-perf.yaml

```diff
@@ -0,0 +1,78 @@
+# MQTT Performance tests pipeline
+# Each platform separated into stages
+# Each stage consists of 
+# - setup job for test host (pull and start 'mqttd' container). See templates/mqtt-perf-setup.yaml
+# - setup job for publisher node and subscriber node (mqtt-benchmark tool writen in go). See templates/mqtt-perf-setup.yaml
+# - a set of test cases. See templates/mqtt-perf-test-case.yaml.
+
+# Pipeline requires custom agent pool (defined in $(pool.name)) 
+# with at least two separate agents:
+# - with 'mqtt-perf-tests-pub' demand - for publisher node
+# - with 'mqtt-perf-tests-sub' demand - for subscriber node
+# And two SSH service connections for AMD64 and ARM32 test hosts:
+#  - $(broker.sshconnection.amd64)
+#  - $(broker.sshconnection.arm32v7)
+# to manage and run 'mqttd' docker container there.
+
+trigger: none
+pr: none
+
+# github repo for our fork of mqtt-benchmark tool.
+resources:
+  repositories:
+  - repository: mqttbenchmark
+    type: github
+    endpoint: Azure/iotedge
+    name: vadim-kovalyov/mqtt-benchmark
+
+variables:
+  benchmarkToolPath: $(Agent.WorkFolder)/mqtt-benchmark
+
+stages:
+################################################################################	
+- stage: linux_arm32v7
+################################################################################	
+  variables:
+    arch: arm32v7
```

Comment:
I believe these variables are used inside templates.  Currently we define template parameter to pass values, I would like to check with @damonbarry if we prefer template parameters than global variables or vice versa.

Reply:
I actually followed the same approach we have in e2e tests.

Reply:
Did you check with Damon already?  it is ok to address it in another PR.

Reply:
The pattern I've seen is that template parameters are added when there are new/additional arguments to pass into a template, and arguments from higher levels are simply used throughout when it's appropriate.

@philipktlin Which of these template parameters are you wondering about? It makes sense to me that we'd map the different platform- or scenario-specific higher-level parameters into parameters for a particular template. It avoids the need to do a bunch of platform-dependent conditional statements down inside the template...

Reply:
Sorry for confusion.  My question is that we didn't pass these variables explicitly to template (for example, line 54-55).  Should we explicitly pass to template?

Reply:
L54-55 define `brokerImageName` and `brokerAddress`. These parameters are assigned different arguments in arm32v7, amd64, and mosquitto control pipelines. If we don't declare parameters here, then the template will need conditional statements with different behavior for each global parameter, reducing its benefit. If we ever added another pipeline, we'd also have to update the code inside the template, which means the abstraction didn't do its job.

Reply:
L54-55 define "- template: templates/mqtt-perf-setup.yaml" and "- template: templates/mqtt-perf-run.yaml".  @damonbarry do you think we should explicitly define parameters for these template usage, e.g. for arch, brokerImageName variables etc.?

Reply:
OK I'm with you now...I was looking at an older revision of the code. 😉 

We had a similar conversation in [one](https://github.com/Azure/iotedge/pull/1477#discussion_r313124325) of my previous PRs. You certainly could define these parameters inside the template instead of (or in addition to) here, but it makes the template WAY more verbose. Also, you have to either (1) come up with (un)reasonable defaults for each template parameter, or (2) assign the empty string, which seems awkward.

A good example of this is `builds/e2e/templates/longhaul-deploy-windows.yaml`: all the parameters at the top are a form of documentation which is kind of nice, but IMO not nice enough to justify all the extra lines. A reference to a variable (whether or not the variable is "declared" in the same file) is, in itself, a form of documentation, especially if you choose decent names. So I'd prefer the simpler form.

Reply:
As discussed with @philipktlin offline, I'll add `brokerSshConnection` as a template parameter (since this is the only variable that I need to use to avoid conditional logic). And to keep things simple, I'll keep the rest as variables in order not to propagate them through nested templates.

Also, as discussed, I'll do it in a separate PR.